api/v1_claim_rewards.go

```diff
@@ -225,65 +206,169 @@ func getValidatorAttestation(args GetValidatorAttestationParams) (*SenderAttesta
 }
 
 // Gets reward claim attestations from AAO and Validators in parallel.
+// Now handles validator selection internally with retry logic for failures.
 func fetchAttestations(
 	ctx context.Context,
 	rewardClaim RewardClaim,
-	validators []string,
+	allValidators []config.Node,
+	excludedOperators []string,
 	antiAbuseOracle config.Node,
 	signature string,
 	hasAntiAbuseOracleAttestation bool,
+	minVotes int,
 ) ([]SenderAttestation, error) {
 
+	// Shuffle the validators for random selection
+	shuffled := slices.Clone(allValidators)
+	rand.Shuffle(len(shuffled), func(i, j int) {
+		shuffled[i], shuffled[j] = shuffled[j], shuffled[i]
+	})
+
+	// Track which operators we've already used (including excluded ones)
+	usedOperators := make(map[string]bool)
+	for _, excluded := range excludedOperators {
+		usedOperators[excluded] = true
+	}
+
+	// Track which specific validators we've marked as bad
+	badValidators := make(map[string]bool)
+
 	offset := 0
 	if !hasAntiAbuseOracleAttestation {
 		offset = 1
 	}
-	attestations := make([]SenderAttestation, len(validators)+offset)
 
-	ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
+	// We need minVotes - len(excludedOperators) successful validator attestations
+	validatorsNeeded := minVotes - len(excludedOperators)
+	attestations := make([]SenderAttestation, 0, validatorsNeeded+offset)
+
+	ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
 	defer cancel()
-	innerGroup, _ := errgroup.WithContext(ctx)
+
+	// Start AAO attestation in parallel if needed
+	var aaoAttestation *SenderAttestation
+	var aaoErr error
+	aaoDone := make(chan struct{})
 
 	if !hasAntiAbuseOracleAttestation {
-		innerGroup.Go(func() error {
+		go func() {
+			defer close(aaoDone)
 			aaoClaim := rewardClaim.RewardClaim
 			aaoClaim.AntiAbuseOracleEthAddress = ""
 			getAntiAbuseAttestationParams := GetAntiAbuseOracleAttestationParams{
 				Claim:                   aaoClaim,
 				Handle:                  rewardClaim.Handle,
 				AntiAbuseOracleEndpoint: antiAbuseOracle.Endpoint,
 			}
-			aaoAttestation, err := getAntiAbuseOracleAttestation(getAntiAbuseAttestationParams)
-			if err != nil {
-				return err
-			}
-			attestations[0] = *aaoAttestation
-			return nil
-		})
+			aaoAttestation, aaoErr = getAntiAbuseOracleAttestation(getAntiAbuseAttestationParams)
+		}()
+	} else {
+		close(aaoDone)
 	}
 
-	for i, validator := range validators {
-		innerGroup.Go(func() error {
-			getValidatorAttestationParams := GetValidatorAttestationParams{
-				Validator:      validator,
-				Claim:          rewardClaim.RewardClaim,
-				UserEthAddress: rewardClaim.RecipientEthAddress,
-				Signature:      signature,
+	// Get validator attestations in parallel rounds
+	successfulValidators := 0
+	currentIndex := 0
+
+	for successfulValidators < validatorsNeeded && currentIndex < len(shuffled) {
+		// Select up to validatorsNeeded validators for this round
```

Comment:
suspiciously many comments... ai write this? 😝 

some are helpful, others seem a bit redundant, but guess it doesn't hurt anything

Reply:
yeah sorry for the slop! i cleaned up some